low-level/sensors/impl/generic/raid_integrity_data.py

```diff
@@ -166,22 +165,27 @@ def _raid_health_monitor(self):
                 # Check RAID device array state is 'idle' or not
                 result = self._check_raid_state(device)
                 if result == "failed":
+                    # Send alert
+                    self.alert_type = self.FAULT
+                    self._alert_msg = RaidAlertMsgs.STATE_MSG.value
+                    self._send_json_msg(self.alert_type, device, self._alert_msg)
+                    # Retry to check RAID state
                     self._retry_execution(self._check_raid_state, device)
                 logger.debug("'idle' state is found in Raid device:{}."
                              .format(device))
 
                 # Check Mismatch count in RAID device files.
                 result = self._check_mismatch_count(device)
                 if result == "failed":
+                    # Send alert
+                    self.alert_type = self.FAULT
+                    self._alert_msg = RaidAlertMsgs.MISMATCH_MSG.value
+                    self._send_json_msg(self.alert_type, device, self._alert_msg)
```

Comment:
Where are limiting the fault alerts? With this patch, frequency to send fault alerts is infact increased, rather than limiting to once. Didnt we decided we will limit fault alert per device once?

Reply:
According to above change, the first time when we get condition result == "failed", it sends the alert and then call the func retry_execution() where it retries every hour until success.